bin/build

```
@@ -0,0 +1,158 @@
+#!/usr/bin/env bash
+
+# fail fast
+#set -eo pipefail
+
+set -eu
+
+start_time=$(date +%s%N)
+
+# debug
+if [[ -v BUILDPACK_APT_DEBUG ]]; then
+  set -x
+fi
+
+# shellcheck disable=SC1090,SC1091
+. "$(dirname "${BASH_SOURCE[0]}")/util.sh"
+
+layers_dir="$1"
+env_dir="$2/env"
+
+if compgen -G "${env_dir}/*" > /dev/null; then
+  for var in "${env_dir}"/*; do
+    declare "$(basename "${var}")=$(<"${var}")"
+  done
+fi
+
+apt_layer=$layers_dir/apt
+mkdir -p "$apt_layer"
+
+# Store which STACK we are running on in the cache to bust the cache if it changes
+if [ -f "$apt_layer"/STACK ]; then
+  CACHED_STACK=$(cat "$apt_layer/STACK")
+else
+  CACHED_STACK=$CNB_STACK_ID
+fi
+
+# Ensure we store the STACK in the cache for next time.
+echo "$CNB_STACK_ID" > "$apt_layer/STACK"
+
+APT_CACHE_DIR="$apt_layer/cache"
+APT_STATE_DIR="$apt_layer/state"
+APT_SOURCELIST_DIR="$apt_layer/sources"   # place custom sources.list here
+APT_SOURCES="$APT_SOURCELIST_DIR/sources.list"
+APT_VERSION=$(apt-get -v | awk 'NR == 1{ print $2 }')
+
+case "$APT_VERSION" in
+	0* | 1.0*) APT_FORCE_YES=(--force-yes);;
+	*)         APT_FORCE_YES=(--allow-downgrades --allow-remove-essential --allow-change-held-packages);;
+esac
+
+if [ -f "$apt_layer"/Aptfile ] && cmp -s Aptfile "$apt_layer"/Aptfile && [[ $CACHED_STACK == "$CNB_STACK_ID" ]] ; then
```

nit; you can simplify to this (& `[[ ]]` vs `[ ]` for consistency)

```suggestion
if [[ -f "$apt_layer"/Aptfile && $CACHED_STACK == "$CNB_STACK_ID" ]] && cmp -s Aptfile "$apt_layer"/Aptfile; then
```

 also TIL about `cmp`!